### PR TITLE
Adds test_system_clock unit test

### PR DESCRIPTION
**Pull request by laurenchilutti**
**Description**
Adds a unit test for [system_clock](https://github.com/NOAA-GFDL/FMS/blob/master/mpp/include/system_clock.h)

Fixes # (issue)

**How Has This Been Tested?**
Tested on Gaea using fre/bronx-16 and cray/hdf-5 modules with `make check`

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by colingladueNOAA** [comment]
Could you try testing with make distcheck? It more closely emulates the process that will be used by TravisCI to test.

File: test_fms/mpp/Makefile.am [hunk 1]
```diff
@@ -50,21 +51,24 @@ test_mpp_print_memuse_stats_stderr_SOURCES=test_mpp_print_memuse_stats_stderr.F9
...
 EXTRA_DIST = input_base.nml \
   test_mpp_domains2.sh \
   test_mpp_pset2.sh	\
   test_mpp2.sh \
-  test_mpp_memutils_mod.sh
+  test_mpp_memutils_mod.sh \
+  test_system_clock.sh
 
 # Clean up
 CLEANFILES = input.nml *.out*
```
Comment (colingladueNOAA) [inline test_fms/mpp/Makefile.am:74]: Add your created module (include_files_mod.mod) here to clean it up.

Comment (laurenchilutti) [reply test_fms/mpp/Makefile.am]: Thank you - that worked and Travis CI tests pass now.

**Comment by colingladueNOAA** [comment]
@thomas-robinson @wrongkindofdoctor If you get the chance, could you review this PR?

**Pull request merged**